sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -121,91 +143,532 @@ struct SptrsvWrap {
           rhs(rhs_),
           nodes_grouped_by_level(nodes_grouped_by_level_) {
-      KK_REQUIRE_MSG(!BlockEnabled, "Blocks are not yet supported.");
-      KK_REQUIRE_MSG(block_size_ == 0, "Blocks are not yet supported.");
+      KK_REQUIRE_MSG(block_size_ == 0,
+                     "Tried to use blocks with the unblocked Common?");
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    size_type get_block_size() const { return 0; }
+
+    // lset
+    KOKKOS_INLINE_FUNCTION
+    void lset(const size_type row, const scalar_t value) const {
+      lhs(row) = value;
+    }
+
+    // add. y += x
+    KOKKOS_INLINE_FUNCTION
+    static void add(const member_type &team, const scalar_t &x, scalar_t &y) {
+      Kokkos::single(Kokkos::PerTeam(team), [&]() { y += x; });
+      team.team_barrier();
+    }
+
+    // serial add. y += x
+    KOKKOS_INLINE_FUNCTION
+    static void add(const scalar_t &x, scalar_t &y) { y += x; }
+
+    // divide. b /= A
+    KOKKOS_INLINE_FUNCTION
+    static void divide(const member_type &team, scalar_t &b, const scalar_t &A,
+                       scalar_t *) {
+      Kokkos::single(Kokkos::PerTeam(team), [&]() { b /= A; });
+      team.team_barrier();
+    }
+
+    // serial divide. b /= A
+    KOKKOS_INLINE_FUNCTION
+    static void divide(scalar_t &b, const scalar_t &A, scalar_t *) { b /= A; }
+
+    // multiply_subtract. C -= A * B
+    KOKKOS_INLINE_FUNCTION
+    static void multiply_subtract(const scalar_t &A, const scalar_t &B,
+                                  scalar_t &C) {
+      C -= A * B;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    static void copy(const member_type &, scalar_t &, const scalar_t &) {}
+
+    KOKKOS_INLINE_FUNCTION
+    static void copy(scalar_t &, const scalar_t &) {}
+
+    // lget
+    KOKKOS_INLINE_FUNCTION
+    scalar_t &lget(const size_type row) const { return lhs(row); }
+
+    // rget
+    KOKKOS_INLINE_FUNCTION
+    scalar_t rget(const size_type row) const { return rhs(row); }
+
+    // vget
+    KOKKOS_INLINE_FUNCTION
+    scalar_t vget(const size_type nnz) const { return values(nnz); }
+
+    // lhs = (lhs + rhs) / diag (team)
+    KOKKOS_INLINE_FUNCTION
+    static void add_and_divide(const member_type &team, scalar_t &lhs_val,
+                               const scalar_t &rhs_val,
+                               const scalar_t &diag_val) {
+      Kokkos::single(Kokkos::PerTeam(team),
+                     [&]() { lhs_val = (lhs_val + rhs_val) / diag_val; });
+    }
+
+    // lhs = (lhs + rhs) / diag (serial)
+    KOKKOS_INLINE_FUNCTION
+    static void add_and_divide(scalar_t &lhs_val, const scalar_t &rhs_val,
+                               const scalar_t &diag_val) {
+      lhs_val = (lhs_val + rhs_val) / diag_val;
+    }
+
+    // print
+    KOKKOS_INLINE_FUNCTION
+    static void print(const scalar_t &item) { std::cout << item << std::endl; }
+
+    KOKKOS_INLINE_FUNCTION
+    static void print(ArrayType rhs, const int) {
+      std::cout << rhs << std::endl;
+    }
+  };
+
```

Comment:
```suggestion
```
You can't use ``std::cout`` inside device code, I would suggest removing this (I assume it was for debugging?) or replace with printf

Reply:
Done, removed.